products/nativescript/market_helpers.py

```diff
@@ -56,20 +81,84 @@ def get_data_market():
 
         return testing_data
 
+    @staticmethod
+    def get_preserved_data():
+        file_path = os.path.join(Settings.TEST_RUN_HOME, 'results.json')
+        data = None
+        if os.path.isfile(file_path):
+            with open(file_path, "r") as json_file:
+                data = json.load(json_file)
+        else:
+            with open(file_path, "w") as new_file:
+                Log.info("Results.json file created " + new_file)
+
+        return data
+
+    @staticmethod
+    def serialize(obj):
+        return obj.__dict__
+
     @staticmethod
     def preserve_data(record):
-        # from string to dict - some = ast.literal_eval(test1)
-        file_path = os.path.join(Settings.TEST_OUT_HOME, 'results.txt')
-        with open(file_path, "a") as myfile:
-            myfile.write(str(record)+"\n")
+        file_path = os.path.join(Settings.TEST_RUN_HOME, 'results.json')
+        preserved_data = Market.get_preserved_data()
+        temp_sample_status = None
+        record_name = record["name"]
+        original_index = None
+
+        if preserved_data:
+            temp_sample_status = next((x for x in preserved_data if x["name"] == record_name), None)
+            if temp_sample_status is not None:
+                original_index = preserved_data.index(temp_sample_status)
+        else:
+            preserved_data = []
+
+        if temp_sample_status is None:
+            temp_sample_status = {
+                "name": record_name,
+                "core": None,
+                "angular": None,
+                "vue": None
+            }
+
+        if record["flavor"] == "core":
+            temp_sample_status["core"] = Market.serialize(Market.get_flavor_status(record))
+
+        if record["flavor"] == "angular":
```

Comment:
We can use if/else here

Reply:
to me using if else instead of separate if conditions is harder to read and has very little benefit for code performance optimisation.  I am fine either way, I am not very well known in python so I am not sure what the "zen" says about this :)